sui_core/src/authority_server.rs

```diff
@@ -58,331 +77,319 @@ impl AuthorityServer {
+        let port = local_addr.port();
+        info!("Listening to TCP traffic on {host}:{port}");
+        // see https://fly.io/blog/the-tokio-1-x-upgrade/#tcplistener-from_std-needs-to-be-set-to-nonblocking
+        std_listener.set_nonblocking(true)?;
+        let listener =
+            tokio_stream::wrappers::TcpListenerStream::new(TcpListener::from_std(std_listener)?);
+
+        let (tx_cancellation, rx_cancellation) = tokio::sync::oneshot::channel();
+        let service = tonic::transport::Server::builder()
+            .add_service(ValidatorServer::new(self))
+            .serve_with_incoming_shutdown(listener, rx_cancellation.map(|_| ()));
+        let handle = AuthorityServerHandle {
+            tx_cancellation,
+            local_addr,
+            handle: tokio::spawn(service),
+        };
+        Ok(handle)
     }
+}
 
-    async fn handle_batch_streaming<'a, 'b, A>(
-        &'a self,
-        request: BatchInfoRequest,
-        channel: &mut A,
-    ) -> Result<(), SuiError>
-    where
-        A: RwChannel<'b>,
-    {
-        // Register a subscriber to not miss any updates
-        let mut subscriber = self.state.subscribe_batch();
-        let message_end = request.end;
-
-        // Get the historical data requested
-        let (mut items, should_subscribe) = self.state.handle_batch_info_request(request).await?;
-
-        let mut last_seq_sent = 0;
-        while let Some(item) = items.pop_front() {
-            // Remember the last transaction sequence number sent
-            if let UpdateItem::Transaction((seq, _)) = &item {
-                last_seq_sent = *seq;
-            }
-
-            // Send all items back to the client
-            let item = serialize_batch_item(&BatchInfoResponseItem(item));
-            channel
-                .sink()
-                .send(Bytes::from(item))
-                .await
-                .map_err(|_| SuiError::CannotSendClientMessageError)?;
-        }
+#[async_trait]
+impl Validator for AuthorityServer {
+    async fn transaction(
+        &self,
+        request: tonic::Request<BincodeEncodedPayload>,
+    ) -> Result<tonic::Response<BincodeEncodedPayload>, tonic::Status> {
+        let mut transaction: Transaction = request
+            .into_inner()
+            .deserialize()
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
 
-        // No need to send live events.
-        if !should_subscribe {
-            return Ok(());
-        }
+        let mut obligation = VerificationObligation::default();
+        transaction
+            .add_tx_sig_to_verification_obligation(&mut obligation)
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
+        obligation
+            .verify_all()
+            .map_err(|e| tonic::Status::invalid_argument(e.to_string()))?;
+        //TODO This is really really bad, we should have different types for checked transactions
+        transaction.is_checked = true;
+
```

Comment:
Yes. Mind opening an issue and replicating the todo in the places in this file that set this flag?